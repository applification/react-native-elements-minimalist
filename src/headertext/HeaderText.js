import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Text } from 'react-native';

class HeaderTextElement extends Component {
  static propTypes = {
    color: PropTypes.string.isRequired,
  };
  static defaultProps = {
    color: '#000000',
  };

  render() {
    return (
      <Text
        numberOfLines={1}
        style={{
          color: this.props.color,
          paddingLeft: 25,
          paddingBottom: 25,
          fontWeight: 'bold',
          fontSize: 50,
        }}
      >
        {this.props.children}
      </Text>
    );
  }
}

export default HeaderTextElement;
