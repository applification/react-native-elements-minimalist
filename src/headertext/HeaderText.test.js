import React from 'react';
import renderer from 'react-test-renderer';

import HeaderText from './HeaderText';

describe('HEADER TEXT', () => {
  it('Renders with DefaultProps', () => {
    expect(
      renderer.create(<HeaderText>Big Header</HeaderText>)
    ).toMatchSnapshot();
  });

  it('Renders with color prop (red)', () => {
    expect(
      renderer.create(<HeaderText color="red">Red Header</HeaderText>)
    ).toMatchSnapshot();
  });

  it('Always render on a single line', () => {
    const component = renderer
      .create(<HeaderText>Longish Header</HeaderText>)
      .toJSON();
    const { numberOfLines } = component.props;

    expect(numberOfLines).toBe(1);
  });

  it('Always renders with paddingLeft, paddingBottom & paddingRight of 25', () => {
    const component = renderer
      .create(<HeaderText>Padded Header</HeaderText>)
      .toJSON();
    const { paddingLeft, paddingBottom, paddingRight } = component.props.style;
  });

  it('Always renders with a fontsize of 35 in bold', () => {
    const component = renderer
      .create(<HeaderText>Header 50px</HeaderText>)
      .toJSON();
    const { fontSize, fontWeight } = component.props.style;

    expect(fontSize).toBe(35);
    expect(fontWeight).toBe('bold');
  });
});
