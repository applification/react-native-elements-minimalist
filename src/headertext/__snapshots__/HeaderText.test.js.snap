// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADER TEXT Renders with DefaultProps 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    Object {
      "color": "#000000",
      "fontSize": 35,
      "fontWeight": "bold",
      "paddingBottom": 25,
      "paddingLeft": 25,
      "paddingRight": 25,
    }
  }
>
  Big Header
</Text>
`;

exports[`HEADER TEXT Renders with color prop (red) 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    Object {
      "color": "red",
      "fontSize": 35,
      "fontWeight": "bold",
      "paddingBottom": 25,
      "paddingLeft": 25,
      "paddingRight": 25,
    }
  }
>
  Red Header
</Text>
`;
